versions/1.0.0.md

```diff
@@ -614,6 +620,74 @@ This object cannot be extended with additional properties and any properties add
   }
 ```
 
+#### Criterion Object
+
+An object used to specify the context, conditions, and condition types that can be used to prove or satisfy assertions specified in Step Object `successCriteria`, Success Action Object `criteria`, and Failure Action Object `criteria`.
+
+There are three flavors of conditions supported:
+- simple - where basic literals, operators, and loose comparisons are used in combination with [Runtime Expressions](#runtime-expressions).
+- regex - where a regex pattern is applied on the supplied context. The context is defined by a [Runtime Expression](#runtime-expressions).
+- JSONPath - where a JSONPath expression is applied. The root node context is defined by a [Runtime Expression](#runtime-expressions).
+
+##### Literals
+As part of a condition expression, you can use `boolean`, `null`, `number`, or `string` data types.
+
+Type | Literal value
+---|---
+`boolean` | `true` or `false`
+`null` |  `null`
+`number` | Any number format supported in [Data Types](#data-types)
+
+##### Operators
+Operator | Description
+---|---
+`<`| Less than
+`<=`| Less than or equal
+`>`| Greater than
+`>=`| Greater than or equal
+`==`| Equal
+`!=`| Not equal
+`!`| Not
+`&&`| And
+`\|\|`| Or
+`()`| Logical Grouping
+`[]`| Index (0-based)
+`.`| Property de-reference
+
+String comparisons `SHOULD` be case insensitive.
+
+##### Fixed Fields
+Field Name | Type | Description
+---|:---:|---
+<a name="criterionContext"></a>context | `{expression}` | A [runtime expression](#runtime-expressions) used to set the context for the condition to be applied on. If `type` is specified, then the `context` MUST be provided (e.g. `$response.body` would set the context that a JSONPath query expression could be applied to).
+<a name="criterionCondition"></a>condition | `string` | **REQUIRED**. The condition to apply. Conditions can be simple (e.g. `$statusCode == 200` which applies a operator on a value obtained from a runtime expression), or a regex, or a JSONPath expression. For regex and [JSONPath](https://datatracker.ietf.org/doc/draft-ietf-jsonpath-base/21/), the `type` and `context` MUST be specified.
+<a name="criterionType"></a>type | `string` | The type of condition to be applied. If specified, the options allowed are `regex` or `JSONPath`. If omitted, then the condition is assumed to be simple, which at most combines literals, operators and [Runtime Expressions](#runtime-expressions).

```

Comment:
I'm in two minds about this myself - but should be add a `simple` type which is the default type, instead of having an implicit simple type if none is specified?

Reply:
I was very much on the fence here. For completeness, it's probably better to have it as part of allowed values. I doubt, I'd ever write it but no harm in allowing folks to be explicit if they want to.